pkg/internal/controller/controller_test.go

```diff
@@ -104,40 +106,21 @@ var _ = Describe("controller", func() {
 			close(done)
 		})
 
-		It("should wait for each informer to sync", func(done Done) {
```

Comment:
The only thing this test ever tested was that `controller.Start` with a cancelled context doesn't error. With the changes in this PR however it will always error because the context passed from the Controller to WaitForSync is cancelled

Reply:
In that case do you think it's worth having a test case that checks that an error is always returned if the context is cancelled?

Reply:
IMHO the only thing someone who starts a Controller with a cancelled context can realistically expect is that it ends quickly, if it errors or not along the way ends up being an implementation detail. It is not going to result in anything useful.